inlong-sort/sort-formats/format-json/src/main/java/org/apache/inlong/sort/formats/json/canal/CanalJsonEnhancedSerializationSchema.java

```diff
@@ -122,9 +127,16 @@ private static RowType createJsonRowType(DataType physicalDataType, List<Writeab
 
     @Override
     public void open(InitializationContext context) {
-        reuse = new GenericRowData(2 + wirteableMetadataFieldGetter.length);
+        int size = 2 + wirteableMetadataFieldGetter.length;
+        if (typeIndex != -1) {
+            size--;
+        }
```

Comment:
I don't undestand this logic

Reply:
The metadata field  of 'type' has been written as the default value. At this time, the user displays the configured 'type', so the 'type' displayed by the user needs to be removed. At this time, the size of the final written data is reduced by 1.

Reply:
please add a description for the method